src/SQL/Format.hs

```diff
@@ -47,10 +47,17 @@ formatLimit :: Limit -> Text
 formatLimit (LimitAll)     = "limit all"
 formatLimit (Limit number) = "limit " <> pack (show number)
 
+formatGroupBy :: GroupBy -> Text
+formatGroupBy (GroupBy (groups)) =
+  "group by "
+  <> formatCommaSeparated formatExpression (toList groups)
+  <> " "
+
 formatOrderBy :: OrderBy -> Text
 formatOrderBy (OrderBy (orders)) =
   "order by "
   <> formatCommaSeparated formatOrderExpression (toList orders)
+  <> " "
```

Comment:
should we put the spaces before the thing we're formatting perhaps? To avoid trailing spaces.

i.e.
```haskell
" group by"
<> ...
```